src/autoscaler/eventgenerator/generator/evaluator_test.go

```diff
@@ -315,26 +155,59 @@ var _ = Describe("Evaluator", func() {
 						Expect(triggerChan).To(BeSent(triggerArrayGT))
 					})
 					Context("when the appMetrics breach the trigger", func() {
-						BeforeEach(func() {
-							database.RetrieveAppMetricsStub = func(appId string, metricType string, start int64, end int64) ([]*models.AppMetric, error) {
-								return appMetricGTBreach, nil
-							}
-						})
-						It("should send trigger alarm to scaling engine", func() {
-							Eventually(scalingEngine.ReceivedRequests).Should(HaveLen(1))
-							Eventually(logger.LogMessages).Should(ContainElement(ContainSubstring("send trigger alarm to scaling engine")))
+						Context("when the appMetrics are enough", func() {
```

Comment:
we do not need to test not enough metrics for each operator.  The test case with not enough metrics should not be under "when the appMetrics brach the trigger"